### PR TITLE
Mimic json/text/buffer method API from micro

### PR DESCRIPTION
**Pull request by floatdrop**
Just a suggestion about interface.

I would advocate to mimic `json` interface from micro, which takes `req` objects as arguments. This allows to write more complex logic around such functions (like wrap them in `if` or `try...catch`).

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -2,21 +2,23 @@ import {serial as test} from 'ava';
 import got from 'got';
 import micro from 'micro';
 import testListen from 'test-listen';
-import m from '.';
+import access from '.';
```
Comment (kevva) [inline test.js]: I usually use `m` (as in module) in my tests for consistency :). Could you change it back?

**Review by kevva — CHANGES_REQUESTED** [review]
This is the API I wanted to use in the first place, but I don't know if it's that common? Feels like most "middleware" modules (at least in `koa` land etc) does a closure if they have options.

If you want, we can change it in https://github.com/kevva/control-access too.

**Comment by floatdrop** [comment]
@kevva closure for options is fine. I think `control-acess` already implements this interface, so anybody can use it in micro. It can be actually more convenient to have `opts` in closure.

**Comment by kevva** [comment]
Did you mean to close this?

**Comment by floatdrop** [comment]
@kevva yeah, I realized, that with this interface it will be just `control-access`. I can reopen thou.

**Pull request closed without merging**